src/pages/VeSDL/OnChainVote.tsx

```diff
@@ -0,0 +1,254 @@
+  const { t } = useTranslation()
+  const [selectedGauge, setSelectedGauge] = useState<GaugeName | null>(null)
+  const [alertMessage, setAlertMessage] = useState<string>()
+  const [voteWeightToSubmit, setVoteWeightToSubmit] = useState<string>("")
+  const [voteUsed, setVoteUsed] = useState<number | undefined>()
+  const [votes, setVotes] = useState<VotesType>({})
+  const gaugeControllerContract = useGaugeControllerContract(true)
+  const [voteWeightInputError, setVoteWeightInputError] = useState<{
+    hasError: boolean
+    errorText: string
+  }>({ hasError: false, errorText: " " })
+
+  const gaugeNames = useMemo(
+    () =>
+      Object.keys(gauges)
+        .filter((lpAddress) => !gauges[lpAddress]?.isKilled)
+        .map((lpAddress) => {
+          const gaugeName = gauges[lpAddress]?.gaugeName
+          const gaugeAddress = gauges[lpAddress]?.address
+          if (gaugeName && gaugeAddress)
+            return {
+              address: gaugeAddress,
+              gaugeName: gaugeName,
+            } as GaugeName
+        }),
+    [gauges],
+  )
+
+  const getVoteUsed = useCallback(async () => {
+    if (account) {
+      const voteUsed = await gaugeControllerContract?.[
+        "vote_user_power(address)"
+      ](account)
+      setVoteUsed(voteUsed?.toNumber())
+    }
+  }, [account, gaugeControllerContract])
+
+  useEffect(() => {
+    if (account && gaugeControllerContract) {
+      gaugeControllerContract.on(
+        "VoteForGauge",
+        (voteDate, fromAddress, gaugeAddress, voteWeight) => {
+          const gaugeName = gaugeNames.find(
+            (gaugeName) =>
+              gaugeName?.address.toLowerCase() ===
+              (gaugeAddress as string).toLowerCase(),
+          )?.gaugeName
+
+          if (
+            gaugeName &&
+            (fromAddress as string).toLowerCase() === account.toLowerCase()
+          ) {
+            setVotes((currentVoteLists) => ({
+              ...currentVoteLists,
+              [gaugeAddress]: {
+                gaugeName,
+                voteDate: (voteDate as BigNumber).toNumber(),
+                weight: voteWeight as BigNumber,
+              },
+            }))
+            void getVoteUsed()
+          }
+        },
+      )
+    }
+    return () => {
+      gaugeControllerContract?.removeAllListeners()
+    }
+  }, [gaugeControllerContract, gaugeNames, votes, getVoteUsed, account])
+
+  const handleVote = async () => {
+    const voteWeightPercent = parseFloat(voteWeightToSubmit)
+    if (!Number.isNaN(voteWeightPercent)) {
+      try {
+        if (veSdlBalance.isZero()) {
+          setAlertMessage("You need veSDL to vote")
+        } else if (selectedGauge?.address && gaugeControllerContract) {
+          const txn = await gaugeControllerContract.vote_for_gauge_weights(
+            selectedGauge.address,
+            Math.round(voteWeightPercent * 100), //convert percent to bps
+          )
+          void enqueuePromiseToast(1, txn.wait(), "vote")
```

Comment:
does toast work when using `void` instead of `await`?

Reply:
You are correct. We should use `await`